packages/utils/src/lib/filter.ts

```diff
@@ -0,0 +1,24 @@
+export function filterItemRefsBy<T extends { refs: object[] }>(
+  items: T[],
+  refFilterFn?: (item: T['refs'][number]) => boolean,
+) {
+  return filterBy(
+    items
+      .map(item => ({
+        ...item,
+        refs: filterBy(item.refs, refFilterFn),
+      }))
+      // remove item with empty refs
+      .filter(item => item.refs.length),
+  );
+}
+
+export function filterBy<T = unknown>(
```

Comment:
All type parameters are `unknown` by default:

```suggestion
export function filterBy<T>(
```